src/networking.c

```diff
@@ -0,0 +1,374 @@
+/*
+ * This file is part of cc-oci-runtime.
+ *
+ * Copyright (C) 2016 Intel Corporation
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+/** \file
+ *
+ * Networking routines, used to setup networking (currently docker specific)
+ *
+ */
+
+#include <string.h>
+#include <stdbool.h>
+#include <signal.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/stat.h>
+#include <linux/if.h>
+#include <linux/if_tun.h>
+#include <ifaddrs.h>
+#include <arpa/inet.h>
+#include <net/if_arp.h>
+
+#include <glib.h>
+#include <glib/gprintf.h>
+
+#include "oci.h"
+#include "util.h"
+
+#define TUNDEV "/dev/net/tun"
+
+/*!
+ * Request to create a named tap interface
+ *
+ * \param tap \c tap interface name to create
+ *
+ * \return \c true on success, else \c false.
+ */
+static gboolean
+cc_oci_tap_create (const gchar *tap)
+{
+	struct ifreq ifr;
+	int fd;
+
+	if( (fd = open(TUNDEV, O_RDWR)) < 0 ) {
+		g_critical("Failed to open [%s] [%s]", TUNDEV, strerror(errno));
+		return false;
+	}
+
+	memset(&ifr, 0, sizeof(ifr));
+	ifr.ifr_flags = IFF_TAP;
+	g_strlcpy(ifr.ifr_name, tap, IFNAMSIZ);
+
+	if( ioctl(fd, TUNSETIFF, (void *) &ifr) < 0 ) {
+		g_critical("Failed to create tap [%s] [%s]",
+			   tap, strerror(errno));
+		close(fd);
+		return false;
+	}
+
+	if( ioctl(fd, TUNSETPERSIST, 1) < 0 ){
+		g_critical("failed to TUNSETPERSIST [%s] [%s]",
+			   tap, strerror(errno));
+		close(fd);
+		return false;
+	}
+
+	close(fd);
+	return true;
+}
+
+
+static gboolean
+cc_oci_netlink_run(const gchar *cmd_line) {
+	gint exit_status = -1;
+	gboolean ret = false;
+
+	ret = g_spawn_command_line_sync(cmd_line,
+				  NULL,
+				  NULL,
```

Comment:
This code will be replaced by netlink code in the future and does not make sense to expend effort here. These are command that should never fail and the failure itself is indication that things have gone badly wrong.